### PR TITLE
large subset warnings in aperture photometry

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request is to address slow performance for the raw profile plot in aperture photometry when using large subsets.  This does ~two things~ the following to attempt to address this case:
1. ~reorders so the results table appears above the plots.  Any size image/subset that jdaviz currently handles _should_ return the results almost immediately... if this is ever not the case, then we should consider adding a spinner while computing the results.~
2. adds a warning below the dropdown box if the user selected the raw profile AND the selected aperture has an area of over 5000 pixels.  This is a somewhat arbitrary cutoff, but can be adjusted if reviewers/users see significant performance degradation at smaller subsets.

https://user-images.githubusercontent.com/877591/199300596-96c72fda-3b0f-4093-9d4f-45c38db1f084.mov

Unfortunately a spinner for just the plot isn't obviously feasible as the drawing is handled in bqplot and the browser... although it might be possible to implement from javascript callbacks if we want to explore that further.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:115]: I have a feeling that this number actually depends on the hardware. Is "5000" representative enough? Should we document how you decided on this number?

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:115]: It likely is, I kind of just guessed, but as I said in the description, I'm hoping that we can revise that during reviews.  Ultimately it isn't a cutoff where we prevent running, just a cutoff where we start displaying the warning, so I think we should err on the side of too low of a number... or maybe we should just always display the message?

**Comment by kecnry** [comment]
> do we want to consider setting a hard limit on subset size as a followup?

We definitely can, but the actual limit probably depends on the memory available to the browser and will vary from machine to machine (I suppose _maybe_ we could detect and estimate that limit that on the javascript-side, or maybe we could request that bqplot handles refusing to overload the memory).

**Comment by pllim** [comment]
Instead of warning, do we just want to exclude listing "large" subsets for Aperture Photometry plugin only?

**Comment by camipacifici** [comment]
How would you define "large"?

**Comment by pllim** [comment]
@camipacifici , currently it is arbitrarily `subset_area > 5000` as stated in the diff.

**Comment by camipacifici** [comment]
I did miss that comment, thanks. Anyway, I meant the motivation to pick a specific number. I would not exclude subsets. The warning seems fine for now. We can revisit this later if we get reports of the app crashing.

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: @larrybradley specifically wanted the fit results to be on top, so moving this down is a step backwards. See https://github.com/spacetelescope/jdaviz/pull/1409#issuecomment-1170358197

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: ok, thanks for that context.  This does keep them with the plot, but moves the plot _and_ the gaussian fit below the photometry results.  If we want to move them both back up to the top, then we might need to look into a javascript solution for detecting when the plot is rendered and show a spinner in the meantime, since the blank plot just suggests something isn't working.

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: Moving things around need PO's blessing, I think. I am neutral on whatever that works.

The warning itself pops up as advertised though it is hard for me to judge whether 5000 is a good cutoff or not.

Comment (camipacifici) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: Maybe this is a Jenn question, but would you consider having two buttons? calculate photometry and plot curve or something like that, so the user can decide not to show either if they do not need it and they would see first what they want.

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: I like that idea, and then the warning is more tied to just the plotting step rather than computing the photometry itself - would you like me to do that as part of this PR?

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: I originally wanted the plotting to be a separate plugin...

I think adding another button would introduce too much diff that goes beyond the scope of this PR. Is there a way to do the warning without moving things around?

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: Yes, I can easily revert changing the order but keep the warning, but see my concerns above for why I think this isn't ideal.  But if the plan is to follow-up (relatively soon) with splitting into two buttons/sections, then the warning on its own is still an incremental improvement.

I'm also curious what @larrybradley thinks - was your original request from the comment @pllim linked just to keep the gaussian results tied closely with the plots, or to have the plot and gaussian results _before_ the photometry results?

Comment (camipacifici) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:162]: Stick with the warning for this PR and open a new one (if you have time now or an issue if you don't) for splitting the button, please.

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:122]:
```suggestion
              Estimated remaining memory: {{predictedMemoryUsagePerc(subset_area).toPrecision(2)}}%
```

Would absolute number be more meaningful? For example: `100 / 500 MB` (if that is indeed in MB)

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue]: I did revert the re-ordering for this PR.  I'm working on some smarter thresholds for the warning and will then open a follow-up ticket for either considering re-ordering, splitting the buttons, or splitting the plugin.

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:122]: Probably, but this is just for internal testing (at least for now).  If we decide to expose this information in the warning, we should think about how best to represent it.

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:193]: Maybe this is more useful? The percentage of *remaining* memory is a moving target. Or even better, get rid of `* 100`.
```suggestion
      return slope * npoints / memory.jsHeapSizeLimit * 100

```

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:193]: Also hard for me to determine whether this is accurate or not unless it also reports actual memory used in generating the plot.

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue]: (moved to #1821)

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue
Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue]: (moved to #1821)

**Review by pllim — APPROVED** [review]
As discussed, this is a good minimal implementation of the request. We can fancy it up later.

Works on dark mode! 🎉

**Comment by pllim** [comment]
I'll let you decide if a change log is needed.

**Comment by kecnry** [comment]
Woops, change log got lost when I removed the re-ordering commit.  Re-added and will merge once CI passes.  Thanks for the reviews!

**Pull request merged**
